drivers/regulator/regulator_cp9314.c

```diff
@@ -0,0 +1,530 @@
+		if (ret < 0) {
+			return ret;
+		}
+		break;
+	default:
+		LOG_ERR("Invalid CP9314 REV:0x%x\n", value);
+		return -ENOTSUP;
+	}
+
+	ret = regulator_cp9314_otp_init(dev);
+	if (ret < 0) {
+		return ret;
+	}
+
+	if (config->en_pin.port != NULL) {
+		if (!gpio_is_ready_dt(&config->en_pin)) {
+			return -ENODEV;
+		}
+
+		ret = gpio_pin_configure_dt(&config->en_pin, GPIO_OUTPUT_INACTIVE);
+		if (ret < 0) {
+			return ret;
+		}
+	}
+
+	ret = i2c_reg_update_byte_dt(&config->i2c, CP9314_REG_CTRL4, CP9314_FRC_OP_MODE,
+				     CP9314_FRC_OP_MODE);
+	if (ret < 0) {
+		return ret;
+	}
+
+	ret = i2c_reg_read_byte_dt(&config->i2c, CP9314_REG_STS_PIN_ADC_0, &value);
+	if (ret < 0) {
+		return ret;
+	}
+
+	value = FIELD_PREP(CP9314_STS_ADDR_LVL, value);
+
+	switch (value) {
+	case CP9314_DEVICE_MODE_HOST_4GANG_0x78:
+	case CP9314_DEVICE_MODE_HOST_4GANG_0x72:
+	case CP9314_DEVICE_MODE_HOST_3GANG_0x78:
+	case CP9314_DEVICE_MODE_HOST_3GANG_0x72:
+	case CP9314_DEVICE_MODE_HOST_2GANG_0x78:
+	case CP9314_DEVICE_MODE_HOST_2GANG_0x72:
+		data->sync_role = CP9314_ROLE_HOST;
+		break;
+	case CP9314_DEVICE_MODE_HOST_STANDALONE_0x78:
+	case CP9314_DEVICE_MODE_HOST_STANDALONE_0x72:
+		data->sync_role = CP9314_ROLE_STANDALONE;
+		break;
+	case CP9314_DEVICE_MODE_DEVICE_4_0x68:
+	case CP9314_DEVICE_MODE_DEVICE_4_0x54:
+		data->sync_role = CP9314_ROLE_DEV4;
+		break;
+	case CP9314_DEVICE_MODE_DEVICE_3_0x56:
+	case CP9314_DEVICE_MODE_DEVICE_3_0x53:
+		data->sync_role = CP9314_ROLE_DEV3;
+		break;
+	case CP9314_DEVICE_MODE_DEVICE_2_0x79:
+	case CP9314_DEVICE_MODE_DEVICE_2_0x73:
+		data->sync_role = CP9314_ROLE_DEV2;
+		break;
+	default:
+		return -EINVAL;
+	}
+
+	if (data->sync_role != CP9314_ROLE_STANDALONE) {
+		ret = cp9314_cfg_sync(dev);
+		if (ret < 0) {
+			return ret;
+		}
+	}
+
+	ret = i2c_reg_update_byte_dt(&config->i2c, CP9314_REG_CTRL4, CP9314_MODE_MASK,
+				     config->initial_op_mode_idx);
+	if (ret < 0) {
+		return ret;
+	}
+
+	regulator_common_data_init(dev);
+
+	return regulator_common_init(dev, false);
+}
+
+static const struct regulator_driver_api api = {
+	.enable = regulator_cp9314_enable,
+	.disable = regulator_cp9314_disable,
+};
+
+#define REGULATOR_CP9314_DEFINE(inst)                                                              \
+	static struct regulator_cp9314_data data_##inst;                                           \
+                                                                                                   \
+	static const struct regulator_cp9314_config config_##inst = {                              \
+		.common = REGULATOR_DT_INST_COMMON_CONFIG_INIT(inst),                              \
+		.i2c = I2C_DT_SPEC_INST_GET(inst),                                                 \
+		.en_pin = GPIO_DT_SPEC_INST_GET_OR(inst, cirrus_en_gpios, {}),                     \
+		.initial_op_mode_idx =                                                             \
+			DT_INST_ENUM_IDX(inst, cirrus_initial_switched_capacitor_mode) + 1U,       \
```

Comment:
You can handle an omitted devicetree entry like this:

```suggestion
			DT_INST_ENUM_IDX_OR(inst, cirrus_initial_switched_capacitor_mode, -1) + 1,       \
```

Is writing zero to select the resistor valid for this device?
If not you can add a condition to the writing of the mode mask.

Reply:
0 is reserved. I'll add a conditional.